src/mod_openssl.c

```diff
@@ -176,6 +178,68 @@ ssl_info_callback (const SSL *ssl, int where, int ret)
     }
 }
 
+// https://wiki.openssl.org/index.php/Manual:SSL_CTX_set_verify(3)#EXAMPLES
+static int
+verify_callback(int preverify_ok, X509_STORE_CTX *ctx)
+{
+    char buf[256];
+    X509 *err_cert;
+    int err, depth;
+    SSL *ssl;
+    handler_ctx *hctx;
+    server *srv;
+
+    err_cert = X509_STORE_CTX_get_current_cert(ctx);
+    err = X509_STORE_CTX_get_error(ctx);
+    depth = X509_STORE_CTX_get_error_depth(ctx);
+
+    /*
+     * Retrieve the pointer to the SSL of the connection currently treated
+     * and the application specific data stored into the SSL object.
+     */
+    ssl = X509_STORE_CTX_get_ex_data(ctx, SSL_get_ex_data_X509_STORE_CTX_idx());
+    hctx = (handler_ctx *) SSL_get_app_data(ssl);
+    srv = hctx->srv;
+
+    X509_NAME_oneline(X509_get_subject_name(err_cert), buf, 256);
+
+    /*
+     * Catch a too long certificate chain. The depth limit set using
+     * SSL_CTX_set_verify_depth() is by purpose set to "limit+1" so
+     * that whenever the "depth>verify_depth" condition is met, we
+     * have violated the limit and want to log this error condition.
+     * We must do it here, because the CHAIN_TOO_LONG error would not
+     * be found explicitly; only errors introduced by cutting off the
+     * additional certificates would be logged.
+     */
+    if (depth > hctx->conf.ssl_verifyclient_depth) {
+        preverify_ok = 0;
+        err = X509_V_ERR_CERT_CHAIN_TOO_LONG;
+        X509_STORE_CTX_set_error(ctx, err);
+    }
+
+    if (!preverify_ok) {
+        log_error_write(srv, __FILE__, __LINE__, "sdsssdss",
+                        "SSL: verify error:num=", err, ":",
+                        X509_verify_cert_error_string(err), ":depth=", depth,
+                        ":", buf);
+    } else {
+        log_error_write(srv, __FILE__, __LINE__, "sdss", "SSL: depth=", depth,
```

Comment:
FYI: I am not a fan of the log_error_write() custom interface either.  Still, you can use (e.g. "SDSS") (capital letters) to avoid the extra space added when you use lowercase (e.g. "sdss") in the custom string format.